client/my-sites/comments/comment-list/index.jsx

```diff
@@ -27,53 +34,98 @@ export class CommentList extends Component {
 
 	componentWillReceiveProps( nextProps ) {
 		if ( ! this.props.comments.length ) {
-			this.setState( { comments: nextProps.comments } );
+			this.setState( { comments: keyBy( nextProps.comments, 'ID' ) } );
+		}
+	}
+
+	showNotice = ( commentId, newStatus, previousStatus ) => {
+		const {
+			createNotice: createCommentNotice,
+			removeNotice: removeCommentNotice,
```

Comment:
personally I'm a fan of changing the `import` so we can leave the props as-is without shadowing concerns. actually I'm not a fan of forbidding shadowing :wink:

if _this_ is the interface (props) then it would make sense to fix the interface and change the periphery